addon/index.ts

```diff
@@ -1,7 +1,12 @@
 import { setComponentManager } from '@ember/component';
+import { setOwner } from '@ember/application';
 
 class SparklesComponent<T = object> {
-  constructor(public args: T) {}
+  constructor(public args: T, owner: any | undefined) {
```

Comment:
This is swapped now I believe

Reply:
should be
```suggestion
  constructor(public args: T, owner?: {}) {
```
to describe the following constraints:
(a) `undefined` need not be passed explicitly under any circumstances
(b) if an argument is passed, it cannot be `null` or `undefined`

Reply:
@mike-north - Seems good, should I use `object` or `{}`? (mostly just curious to the reasoning)

Reply:
@rwjblue 
```ts
let x: object = '';  // ❌
let x: {} = '';  // ✅
```
If you want to allow `string`, `number` `Symbol` and `boolean`, use `{}`
If you just want things that are assignable to JS `Objects`, use `object`